providers/azure/aci.go

```diff
@@ -200,13 +230,89 @@ func NewACIProvider(config string, rm *manager.ResourceManager, nodeName, operat
 	return &p, err
 }
 
+func (p *ACIProvider) setupNetworkProfile(auth *client.Authentication) error {
+	c, err := network.NewClient(auth)
+	if err != nil {
+		return fmt.Errorf("error creating azure networking client: %v", err)
+	}
+
+	var createSubnet bool
+
+	subnet, _ := c.GetSubnet(p.resourceGroup, p.vnetName, p.subnetName)
+	if subnet.ID == "" {
+		createSubnet = true
+	} else {
+		if p.subnetCIDR != subnet.Properties.AddressPrefix {
+			return fmt.Errorf("found existing subnet with different CIDR")
+		}
+		createSubnet = true
```

Comment:
check the delegations here: if the subnet is already delegated to ACI, we don't need to update it.